### PR TITLE
Add getters for most meta-information in mod.

### PR DESCRIPTION
**Pull request by Holt59**
Fixes point 1 of https://github.com/ModOrganizer2/modorganizer/issues/1139

Related `uibase` PR: https://github.com/ModOrganizer2/modorganizer-uibase/pull/83

There are lots of changes but nothing major:

- I renamed a lots of `getX` function to simply `x` to be consistent with other stuffs.
- I moved two `enum`s to `uibase` to be able to access them through `IModInterface`.
- I switch `getCustomUrl()` for `url()` since I don't think the `hasCustomUrl()` is relevant for plugins. Correct me if wrong.

### CODE REVIEW

**Comment by Al12rs** [comment]
This will cause issues with the category_revamp branch where get style code is used in the new changes

**Comment by Holt59** [comment]
> 
> 
> This will cause issues with the category_revamp branch where get style code is used in the new changes

@Silarn Do you think this would be complicated to merge? I don't really like having different styles for getters expose through the plugin API.

**Comment by Al12rs** [comment]
The entire category revamp thing should probably be rethought out with more consideration towards plugins in my opinion. I'm not saying throwing out the current work on it, all the Nexus category querying stuff is good but I think we should think through how to offer plugins to define default and updated categories and also (very important) how to not destroy all existing categories setup.

**Comment by Silarn** [comment]
@Al12rs The revamp code allows you to import the categories based on the loaded instance. So it pulls category data from the game that's in the active instance. Furthermore, you're free to map these categories to your existing category structure. So there's no need to lose your current categories if they're custom. The only blocker is that if you've never saved the category structure, you won't have a category structure saved so there's no base. This is solved by having the old category list instantiated on an existing instance if there are no saved categories, I guess.

The entire point of my revamp code was to avoid having any hard-coded mappings at all and to pull those mappings from the Nexus API rather than relying on plugins.

**Comment by Silarn** [comment]
@Holt59 I can update the code. It's not a big issue.

**Comment by Al12rs** [comment]
@Holt59 I see that some of the things that have been moved to uibase from modinfo.h have been removed from there, making modinfo.h contain less duplication but on the other hand it's much harder to look up a complete interface for a modinfo now since it's fragmented part in uibase and part in modinfo.h, neither containing the whole thing. Before modinfo.h basically covered it.

Would you be against keeping the duplicated definitions in the modinfo.h for the sake of better code navigation? modinfo stuff can already be confusing for new developers.

**Comment by Holt59** [comment]
> Would you be against keeping the duplicated definitions in the modinfo.h for the sake of better code navigation? modinfo stuff can already be confusing for new developers.

You mean adding the pure-virtual declaration from `IModInterface` to `ModInfo` in `modinfo.h`? E.g.

```cpp
// From IModInterface
virtual QString gameName() const = 0;
```

**Comment by Al12rs** [comment]
Yes, assuming that is correct to do. I see it's currently being done for absolutePath()

**Comment by Holt59** [comment]
I cleaned `modinfo.h` (a bit more than necessary, but eh... ). I think in the future, some methods should be moved to `IModInterface` (some getters / setters), but moving setters require a bit more thinking to avoid breaking stuff, so I'd rather do that in a different PR.

File: src/modinfo.h [hunk 1]
```diff
@@ -130,415 +117,556 @@ class ModInfo : public QObject, public MOBase::IModInterface
...
+   * @return the ID of this mod on the repository.
    *
-   * @param categoryID id of the category to set
-   * @param active determines wheter the category is assigned or unassigned
-   * @note this function does not test whether categoryID actually identifies a valid category
-   **/
-  virtual void setCategory(int categoryID, bool active) = 0;
+   * @note For Nexus, this is the Nexus ID of the mod.
+   */
+  virtual int modId() const = 0;
```
Comment (Al12rs) [inline src/modinfo.h:299]: I would use `nexusId()` . This is used in the code to determine whether a mod is nexus stuff or not. It shouldn't be used for other repositories. The setter is setNexusId() and I feel modId can be confused to be some internal id for the mod, like the mod Index. nexusId is really clear on the other hand.

Comment (AnyOldName3) [reply src/modinfo.h:299]: If/when we add integration for other mod repositories, we can rename the function then. It'll probably need changing to a pair of repository name and ID when that happens anyway.

Comment (Holt59) [reply src/modinfo.h:299]: There is already a `repository()` function that returns the repository. 

I'll switch this back to `nexusId()` when I get back home.

**Pull request merged**